core/src/main/java/io/confluent/rest/metrics/RestMetricsContext.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2014 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.rest.metrics;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.kafka.common.metrics.MetricsContext;
+
+public class RestMetricsContext implements MetricsContext {
```

Comment:
It's not clear to me how this class is allowed to be subclassed. It's usually an anti-pattern to have a non-final non-abstract class. What are subclasses allowed to override? Everything? This class keeps state around (`contextLabels`). Are there invariants that subclasses need to maintain on this state? Is this class supposed to be thread-safe? Are implementations also supposed to be? If we are allowing subclassing just for code reuse, there's almost no code in here. I'm not sure it's worth it keeping this class open (and having to think about all the above), instead of just closing it and requiring people to just reimplement it if they need different behaviour.